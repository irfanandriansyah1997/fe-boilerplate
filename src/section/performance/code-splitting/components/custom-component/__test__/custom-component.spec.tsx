import { render } from '@testing-library/react';

import CustomComponent from '..';

describe(`Testing Custom Component`, () => {
  it(`Snapshot Testing`, () => {
    const { getByTestId } = render(<CustomComponent />);

    expect(getByTestId(/custom-component/)).toMatchSnapshot();
  });

  it(`Cross Check Content Custom Component`, () => {
    const { getByTestId, queryByTestId } = render(<CustomComponent />);

    expect(queryByTestId(/custom-component/)).toBeInTheDocument();
    expect(getByTestId(/custom-component/).textContent).toBe(
      `This Is Sample Custom Component`
    );
  });
});
