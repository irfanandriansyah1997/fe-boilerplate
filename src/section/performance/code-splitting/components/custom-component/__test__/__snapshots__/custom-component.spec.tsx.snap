// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Custom Component Snapshot Testing 1`] = `
<div
  class="custom-component"
  data-testid="custom-component"
>
  This Is Sample Custom Component
</div>
`;
